README.md

```diff
@@ -51,28 +54,55 @@ import logging
 
 
 # 1. Define a function that accepts hyperparameters and computes the validation error
-def run_pipeline(hyperparameter_a: float, hyperparameter_b: int):
-    validation_error = -hyperparameter_a * hyperparameter_b
-    return validation_error
+def run_pipeline(
+    hyperparameter_a: float, hyperparameter_b: int, architecture_parameter: str
+) -> dict:
+    # Create your model
+    model = MyModel(architecture_parameter)
+
+    # Train and evaluate the model with your training pipeline
+    validation_error, test_error = train_and_eval(
+        model, hyperparameter_a, hyperparameter_b
+    )
+
+    return {  # dict or float(validation error)
+        "loss": validation_error,
+        "info_dict": {
+            "test_error": test_error
+            # + Other metrics
+        },
+    }
 
 
 # 2. Define a search space of hyperparameters; use the same names as in run_pipeline
 pipeline_space = dict(
-    hyperparameter_a=neps.FloatParameter(lower=0, upper=1),
-    hyperparameter_b=neps.IntegerParameter(lower=1, upper=100),
+    hyperparameter_b=neps.IntegerParameter(
+        lower=1, upper=42, is_fidelity=True
+    ),  # Mark 'is_fidelity' as true for a multi-fidelity approach.
+    hyperparameter_a=neps.FloatParameter(
+        lower=0.001, upper=0.1, log=True
+    ),  # If True, the search space is sampled in log space.
+    architecture_parameter=neps.CategoricalParameter(
+        ["option_a", "option_b", "option_c"]
+    ),
```

Comment:
Assuming we have more comprehensive documentation on how to construct or use neps search space, can we replace this example with a `yaml` file equivalent, just to show that the `basic usage` code requirement is quite low from `neps`? 

Of course, we can do this in the main README only if we also link a page where we can write the Pythonic version of the same thing and show more details such as other hyperparameter, use of priors, and grammars.